Instructions for accessing embedded Longhorn dashboard and updated Volume Snapshot page (#302)
harvester/docs: docs/volume/volume-snapshots.md

Description: Take a snapshot for a volume from the Volume page.
---

A volume snapshot represents a snapshot of a volume on a storage system. After creating a volume, you can create a volume snapshot and restore a volume to the snapshot's state. With volume snapshots, you can easily copy or restore a volume's configuration.

## Create volume snapshots

You can create a volume snapshot from an existing volume by following these steps:

1. Go to the **Volumes** page.
1. Choose the volume that you want to take a snapshot of and select **⋮ > Take Snapshot**.

 ![create-volume-snapshot-1](/img/v1.2/volume/create-volume-snapshot-1.png)
1. Enter a **Name** for the snapshot.

 ![create-volume-snapshot-2](/img/v1.2/volume/create-volume-snapshot-2.png)
1. Select  **Create** to finish creating a new volume snapshot.

> 1. You can check the status of this operation and view all volume snapshots from the **Backup & Snapshot > Volume Snapshots** page. When the **State** becomes **Active**, the volume snapshot is ready to use.

Reviewer: You can check the status of this operation and view all volume snapshots belong to the volume from the volume's **Snapshots** tab. When the **Ready To Use** becomes **√**, the volume snapshot is ready to use.
![image](https://github.com/harvester/docs/assets/15064560/b03812d9-fe55-464b-944a-1183d40bc4e7)

---
Currently, the Active field cannot be used to determine whether a volumesnapshot is ready

Reviewer: @LucasSaintarbor

Author: Thanks @futuretea! Updated per [0a31124](https://github.com/harvester/docs/pull/302/commits/0a311248fd3533febdb8e17529850e91102e62e6)

Description: Take a snapshot for a volume from the Volume page.
---

> After creating a volume, you can create a volume snapshot and use a volume snapshot to restore a new volume. Volume snapshots allow you to copy or restore a volume's configuraiton without creating a completely new volume.

Reviewer 2: Slightly updated the introduction according to the K8s doc [here](https://kubernetes.io/docs/concepts/storage/volume-snapshots/), and added more info about how snapshots can be helpful

## Create volume snapshots

You can create a volume snapshot from an existing volume by following these steps:

1. Go to the **Volumes** page.
1. Choose the volume that you want to take a snapshot of and select **⋮ > Take Snapshot**.

 ![create-volume-snapshot-1](/img/v1.2/volume/create-volume-snapshot-1.png)
1. Enter a **Name** for the snapshot.

 ![create-volume-snapshot-2](/img/v1.2/volume/create-volume-snapshot-2.png)
1. Select  **Create** to finish creating a new volume snapshot.

> 1. You can view all VM snapshots and check the status of this operation from the **Backup & Snapshot > VM Snapshots** page. When the **ReadyToUse** status is set to **true**, the snapshot is complete.

Reviewer 2: ```suggestion
1. You can view all VM snapshots and check the status of this operation from the **Backup & Snapshot > VM Snapshots** page. When the **ReadyToUse** status becomes **true**, the snapshot is created successfully.
```

> 1. You can view all VM snapshots and check the status of this operation from the **Backup & Snapshot > VM Snapshots** page. When the **ReadyToUse** status becomes **true**, the snapshot is created successfully.

Reviewer: This description is about `VM Snapshots` but this page is about `Volume Snapshots`. The user cannot see volume snapshots from the `VM Snapshots` page.

:::note

> A recurring snapshot is currently not supported and is tracked via [harvester/harvester#572](https://github.com/harvester/harvester/issues/572)

Reviewer 2: ```suggestion
A recurring snapshot is currently not supported and is tracked via [harvester/harvester#572](https://github.com/harvester/harvester/issues/572).
```

:::

## Restore a new volume from a volume snapshot

You can restore a new volume from an existing volume snapshot by following these steps:

1. Go to the **Backup & Snapshot > Volume Snapshots** page or go to the **Volumes** page and select the **Snapshots** tab.
1. Select **⋮ > Restore**.
![restore-volume-snapshot-1](/img/v1.2/volume/restore-volume-snapshot-1.png)
![restore-volume-snapshot-2](/img/v1.2/volume/restore-volume-snapshot-2.png)
1. Specify the **Name** of the new volume.
![restore-volume-snapshot-3](/img/v1.2/volume/restore-volume-snapshot-3.png)
1. If the source volume is not an image volume, you can select a different **StorageClass**. You can not change the **StorageClass** if the source volume is an image volume.
![restore-volume-snapshot-4](/img/v1.2/volume/restore-volume-snapshot-4.png)
1. Select **Create** to finish restoring a new volume.

### View the replication and health status in Longhorn

> 1. You can view the replication and health status for the new volume from the Longhorn dashboard. The Longhorn dashboard is available directly in the Harvester dashboard, but you must first enable developer tools & features. Go to the **Preference** page.

Reviewer: the replication and health status for volumes are independent of volume snapshots. Prefer to add a new page.
And we should note that we only support using the embedded Longhorn dashboards for debugging and validation purposes.

Author: @futuretea Removing this section and opening a PR to add a page covering how to view replication and health status for volumes via the embedded longhorn dashboard

![restore-volume-snapshot-5](/img/v1.2/volume/restore-volume-snapshot-5.png)

> 1. Select the **Enable Developer Tools & Features**.

Reviewer 2: ```suggestion
1. Check the **Enable Developer Tools & Features** box.
```
